front/components/assistant/conversation/RetrievalAction.tsx

```diff
@@ -24,57 +26,104 @@ export default function RetrievalAction({
+              color="amber"
+              label={
+                relativeTimeFrame
+                  ? "During the last " +
+                    (relativeTimeFrame.duration > 1
+                      ? `${relativeTimeFrame.duration} ${relativeTimeFrame.unit}s`
+                      : `${relativeTimeFrame.unit}`)
+                  : "All time"
+              }
+            />
+          </Tooltip>
+          <Tooltip label={`Query used for semantic search: ${query}`}>
+            <Chip color="slate" label={query ? shortText(query) : "No query"} />
+          </Tooltip>
+        </Chip.List>
       </div>
-      {!retrievalAction.documents ? (
-        <div>
-          <div className="my-2 text-xs font-bold text-element-600">
-            Retrieving...
-          </div>
-          <Spinner size="sm" />
+      <div className="grid grid-cols-[auto,1fr] gap-2">
+        <div className="grid-cols-auto grid items-center">
+          {!retrievalAction.documents ? (
+            <div>
+              <div className="text-xs font-bold text-element-600">
+                Retrieving...
+              </div>
+              <Spinner size="sm" />
+            </div>
+          ) : (
+            <div className="text-xs font-bold text-element-600">
+              <span>Retrieved:</span>
+            </div>
+          )}
+        </div>
+        <div className="row-span-1">
+          {retrievalAction.documents && (
+            <div onClick={() => setDocListVisible(!docListVisible)}>
+              <Chip color="violet">
+                {retrievalAction.documents.length > 0
+                  ? RetrievedDocumentsInfo(retrievalAction.documents)
+                  : "No documents found"}
+              </Chip>
+            </div>
+          )}
         </div>
-      ) : (
-        <div className="mt-2 text-xs font-bold text-element-600">
-          Retrieved:
-          <Chip color="violet" className="ml-2">
-            {retrievalAction.documents.length > 0
-              ? RetrievedDocumentsInfo(retrievalAction.documents)
-              : "No documents found"}
-          </Chip>
+        <div className="col-start-2 row-span-1">
+          {!!retrievalAction.documents?.length && (
+            <Transition
+              show={docListVisible}
+              enter="transition ease-out duration-200 transform"
+              enterFrom="opacity-0 scale-95"
+              enterTo="opacity-100 scale-100"
+              leave="transition ease-in duration-75 transform"
+              leaveFrom="opacity-100 scale-100"
+              leaveTo="opacity-0 scale-95"
+            >
+              <ul className="ml-2 gap-2">
+                {retrievalAction.documents.map((document, i) => {
+                  const provider = providerFromDocument(document);
+                  return (
+                    <li key={i}>
+                      <a
+                        href={document.sourceUrl || ""}
+                        className="front-bold text-element-800"
```

Comment:
shouldn't it be text-xs to be aligned with the text size of the retrieval component?